hathor/simulator/miner/geometric_miner.py

```diff
@@ -75,7 +75,10 @@ def _generate_mining_block(self) -> 'Block':
         except IndexError:
             signal_bits = 0
 
-        return self._manager.generate_mining_block(signal_bits=signal_bits)
+        block = self._manager.generate_mining_block()
+        block.signal_bits = signal_bits
```

Comment:
Shouldn't we use a `BitSignalingService` here as well?

Reply:
The `BitSignalingService` generates a "fixed" set of signal bits that is the same for all generated blocks (assuming the feature states do not change). Conversely, in [some tests](https://github.com/HathorNetwork/hathor-core/blob/master/tests/feature_activation/test_feature_service.py#L35-L68) I define a different signal bits combination for each block, to test specific scenarios, using the `GeometricMiner`.

In other words, the signal bits are mocked manually here in the `GeometricMiner`, since the `BitSignalingService` doesn't allow such behavior.

Reply:
Got it. Thanks for the clarification.